Frontend/src/app/shared/components/device-care/device-care.component.ts

```diff
@@ -52,16 +52,14 @@ export class DeviceCareComponent implements OnInit {
   constructor(private sharedService: SharedService, private modalCtrl: ModalController, public router: Router, private actRoute: ActivatedRoute) {
     this.actRoute.data.subscribe((data) => {
```

Comment:
This subscription should be unsubscribed somewhere. Otherwise it will cause a memory leak.

Reply:
Working on it sir